packages/plugins/lib/Plugin.ts

```diff
@@ -58,6 +58,24 @@ export class Plugin {
     return this.loadModule(this.definition.preserve.recipe).Recipe;
   }
 
+  /*
+   * compiler
+   */
+
+  definesCompiler(): boolean {
+    return !!this.definition.compiler;
+  }
+
+  loadCompiler(): any {
+    if (!this.definesCompiler()) {
+      throw new TruffleError(
+        `Plugin ${this.module} does not define a \`truffle compiler plugin\`.`
+      );
+    }
+
+    return this.loadModule(this.definition.compiler).Compile;
```

Comment:
Should we have error handling for if the `Compile` method is missing?

Reply:
The error handling is added.